inst/qview/qview.R

```diff
@@ -0,0 +1,194 @@
+library(shiny)
+library(tidyverse)
+library(DT)
+
+## Support Functions ####
+
+#' Function to get the user, or a mock user when the user isn't available.
+whoami <- function(session, .mock_user = "UNKNOWN") {
+  if (!is.null(session$user)) {
+    return(session$user)
+  }
+  return(.mock_user)
+}
+
+#' Function to create a cancel button to be rendered in a Data Table.
+#' @param index The index of the Data Table row that the button will be placed in.
+#' @param .ns The namespace function that should be applied to the button's ID.
+#' @details
+#' The function returns the cancel button as raw HTML with a JavaScript statement
+#' embedded. Using raw HTML + JavaScript allows us to embed the button directly into
+#' a DataTable object.
+cancel_button_creator <- function(index, .ns) {
+  return(
+    as.character(
+      actionButton(
+        as.character(index),
+        glue::glue("Cancel"),
+        onclick = paste0('Shiny.onInputChange(\"', .ns("cancel_button"), '\",  this.id)')
+      )
+    )
+  )
+}
+
+## UI ####
+
+#' queueViewOutput - Data Table displaying the jobs currently in an R Simple Queue.
+#' @param id The Shiny ID to be associated with this object.
+#' @importFrom DT dataTableOutput
+#' @export
+queueViewOutput <- function(id) {
+  ns <- NS(id)
+  tagList(
+    DT::dataTableOutput(ns("queue_table"))
+  )
+}
+
+## Server ####
+
+#' queueView - Server-side function to call with callModule()
+#' @param input Shiny input
+#' @param output Shiny output
+#' @param session Shiny session
+#' @param queue_obj An rrsq::RSimpleQueue object that has already been initialized with $new.
+#' @param refresh_interval The frequency with which the UI should poll the queue and update, in milliseconds.
+#' @importFrom rrsq::RSimpleQueue
+#' @export
+queueView <- function(
+    input,
+    output,
+    session,
+    queue_obj = rrsq::RSimpleQueue$new(),
```

Comment:
the order of params in R should be all non-default params first then default, so you'd want .user to be before the queue_obj